### PR TITLE
[SYCL][FPGA] Fix IVDep translation for pointer parameters

### PR DESCRIPTION
**Pull request by AGindinson**
The original implementation of IVDep for kernel closure parameters
(github.com/KhronosGroup/SPIRV-LLVM-Translator/commit/f0e3545e1f0e3545e)
has only resolved array parameters. When the attribute is applied
to a pointer instead, an additional load instruction gets emitted
for each access by the FE compiler. Accesses to different load
results have been treated as different arrays, while still referring
to the same memory. As a consequence, accesses to the same pointer
have been marked into different `!llvm.index.group` metadata nodes -
essentially, this issue is what the new patch aims to resolve.

Additionally, the IR for the test was bumped up to the new LLVM
version.

Signed-off-by: Artem Gindinson <artem.gindinson@intel.com>

### CODE REVIEW

**Comment by AGindinson** [comment]
@MrSidims, @AlexeySachkov, could you please take a look?

**Comment by AlexeySachkov** [comment]
> github.com/KhronosGroup/SPIRV-LLVM-Translator/commit/f0e3545e1f0e3545e

FYI, the best way to reference other commits on GitHub is `[user|org]/repo@hash`: KhronosGroup/SPIRV-LLVM-Translator@f0e3545e1f0e3545e

**Comment by AlexeySachkov** [comment]
Also, the Travis CI seems to be broken (or even disabled?). Tagging @svenh for awareness. This is not a problem of this PR, but might be a blocker for incoming contributions

File: lib/SPIRV/SPIRVReader.cpp [hunk 1]
```diff
@@ -835,6 +835,9 @@ void SPIRVToLLVM::setLLVMLoopMetadata(const LoopInstType *LM,
           continue;
 
         Value *AccessedPointer = GEP->getPointerOperand();
+        if (auto *LI = dyn_cast<LoadInst>(AccessedPointer))
+          if (Value *LoadOp = LI->getPointerOperand())
```
Comment (AlexeySachkov) [inline lib/SPIRV/SPIRVReader.cpp:839]: Do we really need this as an `if` statement? I mean is it expected to encounter a `load` instruction without its pointer operand?

**Comment by AGindinson** [comment]
I've updated the commit message to better reference the past commit; also moved the test into the right directory.

**Review by MrSidims — COMMENTED** [review]
I'm no longer a fan of using SPIR-V for FPGA: it feels like you need to implement a second backend just to translate a single loop attribute...

**Review by MrSidims — APPROVED** [review]

**Comment by keryell** [comment]
> I'm no longer a fan of using SPIR-V for FPGA: it feels like you need to implement a second backend just to translate a single loop attribute...

This is a trade-off. Probably that if Intel and Xilinx HLS backends were using the ToT Clang/LLVM like any sensible software project would do, it would be useless to go through SPIR-V or handled with dirty IR version downgrading.
If you want to experiment the other way, feel free to contribute to https://github.com/triSYCL/sycl/tree/sycl/unified/next/llvm/lib/SYCL ;-)

**Comment by AlexeySachkov** [comment]
I've just enabled GitHub Actions CI - I will try to trigger it by closing and re-opening the PR

**Review by AlexeySotkin — APPROVED** [review]

**Pull request merged**